src/test/java/test/JiraS3EnabledTest.java

```diff
@@ -0,0 +1,85 @@
+package test;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.TestInfo;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.EnumSource;
+import test.helm.Helm;
+import test.model.Product;
+
+import java.util.Map;
+
+import static test.jackson.JsonNodeAssert.assertThat;
+
+class JiraS3EnabledTest {
+    private Helm helm;
+
+    @BeforeEach
+    void initHelm(TestInfo testInfo) {
+        helm = new Helm(testInfo);
+    }
+
+
+    @ParameterizedTest
+    @EnumSource(value = Product.class, names = {"jira"})
+    void jira_s3_avatars_storage_env_vars(Product product) throws Exception {
+        final var resources = helm.captureKubeResourcesFromHelmChart(product, Map.of(
+                product + ".s3AvatarsStorage.bucketName", "my-bucket",
+                product + ".s3AvatarsStorage.bucketRegion", "my-region"
+        ));
+
+        resources.getStatefulSet(product.getHelmReleaseName())
+                .getContainer()
+                .getEnv()
+                .assertHasValue("ATL_S3AVATARS_BUCKET_NAME", "my-bucket")
+                .assertHasValue("ATL_S3AVATARS_REGION", "my-region");
+    }
+
+    @ParameterizedTest
+    @EnumSource(value = Product.class, names = {"jira"})
+    void jira_s3_avatars_storage_endpoint_override(Product product) throws Exception {
+        final var resources = helm.captureKubeResourcesFromHelmChart(product, Map.of(
+                product + ".s3AvatarsStorage.bucketName", "my-bucket",
+                product + ".s3AvatarsStorage.bucketRegion", "my-region",
+                product + ".s3AvatarsStorage.endpointOverride", "http://minio.svc.cluster.local"
+        ));
+
+        resources.getStatefulSet(product.getHelmReleaseName())
+                .getContainer()
+                .getEnv()
+                .assertHasValue("ATL_S3AVATARS_BUCKET_NAME", "my-bucket")
+                .assertHasValue("ATL_S3AVATARS_REGION", "my-region")
+                .assertHasValue("ATL_S3AVATARS_ENDPOINT_OVERRIDE", "http://minio.svc.cluster.local");
+    }
+
+    @ParameterizedTest
+    @EnumSource(value = Product.class, names = {"jira"})
+    void jira_s3_avatars_storage_no_endpoint_override(Product product) throws Exception {
+        final var resources = helm.captureKubeResourcesFromHelmChart(product, Map.of(
+                product + ".s3AvatarsStorage.bucketName", "my-bucket",
+                product + ".s3AvatarsStorage.bucketRegion", "my-region"
+        ));
+
+        resources.getStatefulSet(product.getHelmReleaseName())
+                .getContainer()
+                .getEnv()
+                .assertHasValue("ATL_S3AVATARS_BUCKET_NAME", "my-bucket")
+                .assertHasValue("ATL_S3AVATARS_REGION", "my-region")
+                .assertDoesNotHaveAnyOf("ATL_S3AVATARS_ENDPOINT_OVERRIDE");
```

Comment:
Nit: maybe this test can be removed (kinda a duplicate of `jira_s3_avatars_storage_env_vars`) and to cover the last assertion add `.assertDoesNotHaveAnyOf("ATL_S3AVATARS_ENDPOINT_OVERRIDE");` to `jira_s3_avatars_storage_env_vars`?